internal/api/handler/list_projects_v1alpha1.go

```diff
@@ -0,0 +1,48 @@
+package handler
+
+import (
+	"context"
+
+	"github.com/bufbuild/connect-go"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kubev1alpha1 "github.com/akuity/kargo/api/v1alpha1"
+	svcv1alpha1 "github.com/akuity/kargo/pkg/api/service/v1alpha1"
+)
+
+type ListProjectsV1Alpha1Func func(
+	context.Context,
+	*connect.Request[svcv1alpha1.ListProjectsRequest],
+) (*connect.Response[svcv1alpha1.ListProjectsResponse], error)
+
+func ListProjectsV1Alpha1(
+	kc client.Client,
+) ListProjectsV1Alpha1Func {
+	return func(
+		ctx context.Context,
+		req *connect.Request[svcv1alpha1.ListProjectsRequest],
+	) (*connect.Response[svcv1alpha1.ListProjectsResponse], error) {
+
+		// Only list projects which contain an Environment
+		var list kubev1alpha1.EnvironmentList
+		if err := kc.List(ctx, &list); err != nil {
+			return nil, connect.NewError(connect.CodeInternal, err)
+		}
+
+		var projectMap = make(map[string]bool)
+		var projects []*svcv1alpha1.Project
+		for _, env := range list.Items {
+			if _, ok := projectMap[env.Namespace]; ok {
+				continue
+			}
+			projectMap[env.Namespace] = true
+			projects = append(projects, &svcv1alpha1.Project{
+				Name: env.Namespace,
+			})
+		}
```

Comment:
just thinking if the namespaces are specifically created for kargo projects, these namespaces should be labeled for proper identification and that can be used for listing them. Though that won't work if any random namespace can be used as a project, but I don't see that being very clean imo.

Reply:
As far as I understand this is also necessary because if a project is created but has no env current approach won't work. So a user cannot choose the project they want the env to be created in from the ui if the namespace already doesn't have envs.

Reply:
Completely agree, however, at the moment there is no label documented to distinguish a Kargo project namespace from a random namespace. 

@krancour Correct me if I'm wrong, but a Kargo `Project` is not its own CRD but rather just using a different term for a namespace. Assuming that is the case, I can introduce a new label to indicate Kargo project namespaces and update documentation accordingly.

I propose `kargo.akuity.io/project: "true"`. WDYT @gdsoumya @krancour ?

Reply:
That sounds good to me but will also check in with @krancour and @jessesuen